pkg/agent/controller/networkpolicy/node_reconciler_linux.go

```diff
@@ -0,0 +1,703 @@
+		set = set.Union(ipBlocksToIPNets(rule.To.IPBlocks, isIPv6))
+	}
+	if set.Has("0.0.0.0/0") || set.Has("::/0") {
+		return nil
+	}
+	return set
+}
+
+func getCoreIPTChain(rule *CompletedRule) string {
+	if rule.Direction == v1beta2.DirectionIn {
+		return config.NodeNetworkPolicyIngressRulesChain
+	}
+	return config.NodeNetworkPolicyEgressRulesChain
+}
+
+func buildCoreIPTRule(ipProtocol iptables.Protocol,
+	iptChain string,
+	ipset string,
+	ipNet string,
+	iptRuleTarget string,
+	iptRuleComment string,
+	service *v1beta2.Service,
+	isIngress bool) string {
+	builder := iptables.NewRuleBuilder(iptChain)
+	if isIngress {
+		if ipset != "" {
+			builder = builder.MatchIPSetSrc(ipset)
+		} else if ipNet != "" {
+			builder = builder.MatchCIDRSrc(ipNet)
+		}
+	} else {
+		if ipset != "" {
+			builder = builder.MatchIPSetDst(ipset)
+		} else if ipNet != "" {
+			builder = builder.MatchCIDRDst(ipNet)
+		}
+	}
+	if service != nil {
+		transProtocol := getServiceTransProtocol(service.Protocol)
+		switch transProtocol {
+		case "tcp":
+			fallthrough
+		case "udp":
+			fallthrough
+		case "sctp":
+			builder = builder.MatchTransProtocol(transProtocol).
+				MatchSrcPort(service.SrcPort, service.SrcEndPort).
+				MatchDstPort(service.Port, service.EndPort)
+		case "icmp":
+			builder = builder.MatchICMP(service.ICMPType, service.ICMPCode, ipProtocol)
+		}
+	}
+	return builder.SetTarget(iptRuleTarget).
+		SetComment(iptRuleComment).
+		Done().
+		GetRule()
+}
+
+func buildServiceIPTRules(ipProtocol iptables.Protocol, services []v1beta2.Service, iptChain string, iptRuleTarget string) []string {
+	var rules []string
+	builder := iptables.NewRuleBuilder(iptChain)
+	for _, svc := range services {
+		copiedBuilder := builder.CopyBuilder()
+		transProtocol := getServiceTransProtocol(svc.Protocol)
+		switch transProtocol {
+		case "tcp":
+			fallthrough
+		case "udp":
+			fallthrough
+		case "sctp":
+			copiedBuilder = copiedBuilder.MatchTransProtocol(transProtocol).
+				MatchSrcPort(svc.SrcPort, svc.SrcEndPort).
+				MatchDstPort(svc.Port, svc.EndPort)
+		case "icmp":
+			copiedBuilder = copiedBuilder.MatchICMP(svc.ICMPType, svc.ICMPCode, ipProtocol)
+		}
+		rules = append(rules, copiedBuilder.SetTarget(iptRuleTarget).
+			Done().
+			GetRule())
+	}
+	return rules
+}
+
+func genServiceIPTRuleChain(ruleID string) string {
+	return fmt.Sprintf("%s-%s", prefix, strings.ToUpper(ruleID))
+}
+
+func genIPSetName(ruleID string, isIPv6 bool) string {
+	suffix := "4"
+	if isIPv6 {
+		suffix = "6"
+	}
+	return fmt.Sprintf("%s-%s-%s", prefix, strings.ToUpper(ruleID), suffix)
+}
+
+func ruleActionToIPTTarget(ruleAction *secv1beta1.RuleAction) string {
+	var target string
+	switch *ruleAction {
+	case secv1beta1.RuleActionDrop:
+		target = iptables.DropTarget
+	case secv1beta1.RuleActionReject:
+		target = iptables.RejectTarget
+	case secv1beta1.RuleActionAllow:
+		target = iptables.AcceptTarget
+	default:
+		klog.InfoS("Unknown rule action", "action", ruleAction)
```

Comment:
If the target falls here, it means the target it empty, what's the default behavior of the iptables? could you check if it's what we expect?

Reply:
Will check.

Reply:
Will remove this since this will be never reachable.